src/_common/video/player/shaka.ts

```diff
@@ -29,42 +34,33 @@ export default class AppVideoPlayerShaka extends Vue {
 	private isDestroyed = false;
 	private previousState: ShakaTrack | null = null;
 	private currentState: ShakaTrack | null = null;
-	private videoStartTime = 0;
-
-	$refs!: {
-		video: HTMLVideoElement;
-	};
+	private video?: HTMLVideoElement;
+	// Function to run after setting up Shaka.
+	private syncFunction!: () => void;
 
 	get shouldAutoplay() {
 		return this.autoplay;
 	}
 
-	mounted() {
-		this.init();
-	}
-
 	beforeDestroy() {
-		this.trackPlaytime();
 		this.isDestroyed = true;
 		if (this.shakaPlayer) {
 			this.shakaPlayer.destroy();
 			this.shakaPlayer = undefined;
 		}
 	}
 
-	private async init() {
+	async initShakaWithVideo(video: HTMLVideoElement, syncFunction: () => void) {
```

Comment:
I think instead of syncFunction, the AppVideo should expect this function to return either true or false (Promise<boolean>). This way it would know if the initialization failed or not. Just await it in AppVideo. No need to pass a sync function that we need to remember to call, I think.